client/src/actions/user.constants.js

```diff
@@ -0,0 +1,33 @@
+import { userConstants } from "./constants";
+import axios from "../helpers/axios";
+
+export const signup = (user) => {
+    console.log(user);
+
+    return async (dispatch) => {
+        dispatch({ type: userConstants.USER_REGISTER_REQUEST });
+        const res = await axios.post('/signup', {
+            ...user
+        })
+
+        if ( res.status === 201 ){
```

Comment:
Safe condistion `if(res.data){`

Reply:
if I will directly restructure data as you mention below  `const {data} = await axios...` then where to initialize `res`?